### PR TITLE
Feat: Add support for text mints

### PR DESCRIPTION
**Pull request by kittypurrry**
This PR adds support for ASCII text mints. There are two new options added to the `MintForm` component - _This is a text mint_ checkbox and a _monospace font required_ checkbox.

For typed mints, the cover and thumbnail images are automatically generated based on the text input and the font selected. The description will also automatically be replaced by the text mint input.

**Minting Form**
The textarea renders the monospace (IBM Plex Mono) / non-monospace font (Source Sans Pro) depending on the _monospace font required_ checkbox value.

**_Monospace:_**
image

**_Non-monospace:_**
image

**Preview / Token Page**
_Example: /objkt/850488_

The preview component displays the typed art, in either a monospace font a non-monospace font (Source Sans Pro). If Monospace is selected, a 'monospace' tag is automatically added to ensure that it renders properly on OBJKT; and viceversa.

image

**Feed Page**

Text NFTs are rendered as their cover/thumbnail images on the feed listing page, similar to how OBJKT does it. The actual text mint is only rendered as in the individual token page. 

image

Notes
- Source Sans Pro was selected for text mints as it's the default font (non-mono) on OBJKT. This ensures text are encoded correctly and displayed as intended on OBJKT and Teia. 
- IBM Plex Mono was selected for monospace text mints as it was the font used by typed.art. Currently, it seems that OBJKT does not provide much monospace support for text mints, and a feature request has been raised here: https://roadmap.objkt.com/b/software-feature-request/use-a-monospaced-font-for-text-tokens/. As a result, do expect that some of the monospace typed art would appear broken on OBJKT. However, Teia has taken care of it by rendering the right fonts for your monospace typed art. See an example minted type mint on OBJKT: https://objkt.com/tokens/hicetnunc/850488

--

_Update [21/04]:_
- UltimateTextToImage package has been uninstalled, replace by using node-canvas as done in typed.art._ 
- Moved 'This is a text mint' checkbox to be before description, so that when checked, the description box is automatically hidden and replaced by the text mint input itself.

**Non-text mint**
image

**Text mint checked**
image

_Update [05/05]:_
- Change Mono from IBM Plex Mono to Iosevka

### CODE REVIEW

**Comment by melMass** [comment]
Thanks, almost good for me, besides the comment I just made I have a few questions:
- Why not use ioveska for `.mono`?
- I didn't follow the discussions that lead to this PR so I just want to be sure it was talked about with someone at OBJKT.com?

We can clean it up more in a separate PR, but fI think it's mostly fine for now if the metadata are good for everyone

File: src/utils/mint.ts [hunk 1]
```diff
@@ -114,3 +115,131 @@ export const generateCoverAndThumbnail = async (
...
+        const blob = await res.blob()
+        const file = new File([blob], 'Generated Cover.png', {
+          type: 'image/png',
+        })
+
+        resolve(file)
+      }
+    }
+  })
+}
```
Comment (melMass) [inline src/utils/mint.ts]: Can you simplify/refactor this a bit for clarity? I moved the text.length check first to un-nest it, changed ts types and use only async/await but I'm still not fully getting the logic of it.

**Comment by melMass** [comment]
Regarding the metadata, it follows typedart logic (minus the on chain text) so it's actually fine!

**Comment by melMass** [comment]
When checking the generated previews I get weird results:

https://cache.teia.rocks/ipfs/QmYdXkZM2nCX38gwrdsD21VG1Vq2P4eur1DgDMu3aLocWZ
https://cache.teia.rocks/ipfs/QmXnZrk9u4KUR9bWNem4oQQS6VpuCuS2FXFDosRqExAeww

Another thing is that it's currently not handling single long lines and I'm not sure what we want to happen in these cases:

**input text**
> In the realm of decentralization, power disperses like scattered light,freedom blooms in every node, no central might.Trust in the collective, no single hand's sway, innovation thrives in this distributed array. From governance to finance, it's a paradigm shift, where autonomy reigns, and control takes its lift.

image

**Comment by melMass** [comment]
Thanks Ziroh
There is one edge case that i think needs testing (cc @merchantcoppola) very long text (for the css thing that you just added)

**Review by Zir0h — APPROVED** [review]

**Pull request merged**